macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
 			layer.contentsScale = scale
 		}
 	}
-	
-	/// Tint color of the button content.
-	@objc override public var contentTintColor: NSColor? {
+
+	private var format: ButtonFormat {
 		get {
-			if style == .primaryFilled {
-				return .white
-			} else {
-				return primaryColor
-			}
+			return ButtonFormat(
+				size: self.size,
+				style: self.style,
+				accentColor: self.accentColor
+			)
 		}
 		set {
-			super.contentTintColor = newValue
-			needsDisplay = true
+			self.accentColor = newValue.accentColor
+			self.style = newValue.style  // depends on accentColor
+			self.size = newValue.size
 		}
 	}
 
-	/// The primary color of the button, AKA, the fill color in the primaryFilled style, and the outline in the primaryOutline style
-	@objc public var primaryColor: NSColor = defaultPrimaryColor {
-		didSet {
-			if primaryColor != oldValue {
-				needsDisplay = true
-			}
-		}
-	}
+	/// Foreground text/image color when button is not pressed or disabled.
+	@objc public var contentTintColorRest: NSColor?
 
-	/// The primary style of the button
-	@objc public var style: ButtonStyle = .primaryFilled {
-		didSet {
-			if style != oldValue {
-				if style == .primaryFilled {
-					contentTintColor = .white
-				} else {
-					contentTintColor = primaryColor
-				}
-				needsDisplay = true
-			}
-		}
-	}
+	/// Foreground text/image color when button is pressed.
+	@objc public var contentTintColorPressed: NSColor?
 
-	/// Background color in rest state. Hover and pressed state colors will adjust accordingly using the systemEffect.
-	@objc public lazy var restBackgroundColor: NSColor  = fillColor {
-		didSet {
-			if restBackgroundColor != oldValue {
-				needsDisplay = true
+	/// Foreground text/image color when button is disabled.
+	@objc public var contentTintColorDisabled: NSColor?
+
+	/// Button foreground text/image color.  Setting this property determines the colors for the resting
+	/// (normal), pressed and disabled states by applying system effects.  The value returned when getting
+	/// this property will depend on which state currently applies to the button.  To set the state-specific
+	/// colors separately, use the `contentTintColorRest`, `contentTintColorPressed`,
+	/// and/or `contentTintColorDisabled` properties.
+	open override var contentTintColor: NSColor? {
+		get {
+			if !isEnabled {
+				return contentTintColorDisabled
+			} else if mouseDown {
+				return contentTintColorPressed
+			} else {
+				return contentTintColorRest
 			}
 		}
-	}
-	
-	private func disabledColor(for color: NSColor) -> NSColor {
-		return color.withSystemEffect(.disabled)
+		set {
+			contentTintColorRest = newValue
+			contentTintColorPressed = newValue == .clear ? newValue : newValue?.withSystemEffect(.pressed)
```

Comment:
Just curious, what happens if you apply the pressed system effect to .clear?

Reply:
.clear is black with 100% alpha.  Applying the .pressed system effect removes some of the alpha, making it shade the background.  This generally does not make visual sense, because it adds visible borders and backgrounds to buttons that are not supposed to have them.

This is an artifact of .clear being a degenerate point in the RGBA space, and applying the system effect would look different based on the choice of RGB value in .clear, even though those values should be irrelevant.

Reply:
Interesting. I wonder if that's a peculiarity that's worth being an Apple Bug? Not related to the review though